content/blog/non-decimal-numbers-in-tech/index.md

```diff
@@ -0,0 +1,225 @@
+Take the following example:
+
+![A "X" on the two column and a "X" on the ones column which add together to make 3](./base_2_3_symbols.svg)
+
+In this example, both `1` and `2` are present, so we add them together to make **`3`**. You'll see that since we can only have a value present or not present — because we only have two symbols in binary — this conversion has less steps than using decimal. For example, if you only wanted the number two, you could simply mark the `1` as "not present" using the **`O`**
+
+![A "X" on the two column and an "O" on the ones column which add together to make 2](./base_2_2_symbols.svg)
+
+You can even replace the two symbols with `1` and `0` to get the actual binary number of `10` in order to represent `2`
+
+![A "1" on the two column and an "0" on the ones column which add together to make 2](./base_2_2.svg)
+
+So how does this play out when trying to represent the number **`50`** in binary?
+
+![The binary number "110010" which shows 32, 64, and 2 to combine together to make 50. See the below explaination for more](./base_2_50.svg)
+
+As you can see, we create columns that are exponents of the number `2` for similar reasons as exponents of `10`; You can't represent `4`, `8`, `16`, or `32` without creating a new column otherwise.
+
+> Remember, in this system a number can only be present or not, there is no _`2`_. Because of this, without the **`4`** column, there can only be a `1` and a `2`, which makes up **`3`**. Continuing on with this pattern: without an **`8`** column, you can only have a `4`, `2`, and `1` which would yeild you **`7`**.
+
+Once each of these exponents is laid out, we can start adding 1s where we have the minimum amount of value. EG:
+Is `64` less than or equal to `50`? No. That's a **`0`**
+Is `32 <= 50`? Yes, therefore that's a **`1`**
+`50 - 32 = 18`
+Moving down the list, is `16 <= 18`? Yes, that's a **`1`**
+`18 - 16 = 2`
+Is `8 <= 2`? No, that's a **`0`**
+`4 <= 2`? No, that's a **`0`** as well
+`2 <= 2`? Yes, that's a **`1`**
+`2 - 2` is **`0`**. That means every number afterwards (in this case only `1` is left) is not present, therefore is a **`0`**.
+
+Add up all those numbers:
+|Column|Value|
+|--|--|
+|`64`| **`0`**|
+|`32`| **`1`**|
+|`16`| **`1`**|
+|`8`| **`0`**|
+|`4`| **`0`**|
+|`2`| **`1`**|
+|`1`| **`0`**|
+
+And voilà, you have the binary representation of `50`: **`0110010`**
+
+> Author's note:
+>
+> While there are plenty of ways to find the binary representation of a decimal number, this example uses a "greedy" alogrithm. I find this algorithm to flow the best with learning of the binary number system, but it's not the only way (or even the best way, oftentimes).
```

Comment:
Huge thanks to @tamuseanmiller for reminding me of the name of this algorithm and suggesting to include it for reference. Was a great suggestion to add it!